### PR TITLE
Reworked DLC sequence numbers to disable RBF

### PR DESCRIPTION
**Pull request by nkohen**
Using `0xffffffffe` as the sequence numbers on non-CSV inputs in DLC txs to opt-out of RBF (see BIP 125).

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/protocol/transaction/TransactionConstants.scala [hunk 1]
```diff
@@ -11,6 +11,7 @@ trait TransactionConstants {
   lazy val validLockVersion = Int32(2)
   lazy val lockTime = UInt32.zero
   lazy val sequence = UInt32(4294967295L)
+  lazy val disableRBFSequence = UInt32(4294967294L)
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/protocol/transaction/TransactionConstants.scala]: nit: maybe make this `sequence - 1` so it is clearer to see that it is `MAX - 1`

File: core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala [hunk 2]
```diff
@@ -619,12 +616,10 @@ object TxBuilder {
               //none of these script types affect the sequence number of a tx
               //the sequence only needs to be adjustd if we have replace by fee (RBF) enabled
               //see BIP125 for more information
-              val sequence =
-                if (isRBFEnabled) UInt32.zero else TransactionConstants.sequence
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala]: This comment is no longer relevant

File: core/src/main/scala/org/bitcoins/core/protocol/transaction/TransactionConstants.scala (hunk 1 above)
Comment (Christewart) [reply core/src/main/scala/org/bitcoins/core/protocol/transaction/TransactionConstants.scala]: This enables RBF: https://github.com/bitcoin/bips/blob/master/bip-0125.mediawiki#summary


>Explicit signaling: A transaction is considered to have opted in to allowing replacement of itself if any of its inputs have an nSequence number less than (0xffffffff - 1).

File: core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala [hunk 3]
```diff
@@ -570,7 +570,7 @@ object TxBuilder {
     */
   def calcSequenceForInputs(
       utxos: Seq[UTXOSpendingInfo],
-      isRBFEnabled: Boolean): Seq[TransactionInput] = {
+      defaultSequence: UInt32 = UInt32.zero): Seq[TransactionInput] = {
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala]: why not make this `Policy.sequence` or something like that? I think users should have to explicitly enable things like RBF/CSV

**Pull request merged**